### PR TITLE
Improve loading of Ruby scripts when using the src attribute

### PR DESCRIPTION
**Pull request by ledsun**
- Parallelization of downloads
- Error response handling
- URL encoding

### CODE REVIEW

File: packages/npm-packages/ruby-wasm-wasi/src/browser.script.ts [hunk 1]
```diff
@@ -16,17 +16,35 @@ export const main = async (pkg: { name: string; version: string }) => {
 };
 
 const runRubyScriptsInHtml = async (vm) => {
```
Comment (kateinoigakukun) [inline packages/npm-packages/ruby-wasm-wasi/src/browser.script.ts:18]: It's not directly related to your change, but we need to consider when the `text/ruby` scripts should be scanned.
[`DOMContentLoaded`](https://developer.mozilla.org/en-US/docs/Web/API/Document/DOMContentLoaded_event) looks more suitable to me than the current immediate scan.

Could you update it to defer script evals after the event if you don't mind?

File: packages/npm-packages/ruby-wasm-wasi/src/browser.script.ts [hunk 2]
```diff
@@ -16,17 +16,35 @@ export const main = async (pkg: { name: string; version: string }) => {
...
+  // see: https://html.spec.whatwg.org/multipage/scripting.html#inline-documentation-for-external-scripts
+  if (tag.hasAttribute("src")) {
+    const url = encodeURI(tag.getAttribute("src"));
+    const response = await fetch(url);
+
+    if (response.ok) {
+      return await response.text();
     }
+
+    return Promise.resolve("");
```
Comment (kateinoigakukun) [inline packages/npm-packages/ruby-wasm-wasi/src/browser.script.ts:46]: nits: I prefer `null` to an empty string to indicate no data

```suggestion
    return Promise.resolve(null);
```

File: packages/npm-packages/ruby-wasm-wasi/src/browser.script.ts [hunk 3]
```diff
@@ -16,17 +16,35 @@ export const main = async (pkg: { name: string; version: string }) => {
...
+    }
+  }
+};
+
+const loadScriptAsync = async (tag: Element): Promise<string> => {
+  // Inline comments can be written with the src attribute of the script tag.
+  // The presence of the src attribute is checked before the presence of the inline.
+  // see: https://html.spec.whatwg.org/multipage/scripting.html#inline-documentation-for-external-scripts
+  if (tag.hasAttribute("src")) {
+    const url = encodeURI(tag.getAttribute("src"));
```
Comment (kateinoigakukun) [inline packages/npm-packages/ruby-wasm-wasi/src/browser.script.ts:39]: Should you add a rationale for the encoding? I couldn't find the reason, sry.

File: packages/npm-packages/ruby-wasm-wasi/src/browser.script.ts (hunk 1 above)
Comment (ledsun) [reply packages/npm-packages/ruby-wasm-wasi/src/browser.script.ts:27]: Delayed execution of runRubyScriptsInHtml at https://github.com/ruby/ruby.wasm/pull/53/commits/3c4ef9e253914225e21d9c49f07287001e084eae .
Also, added support for cases where the DOMContentLoaded event has already been fired.

File: packages/npm-packages/ruby-wasm-wasi/src/browser.script.ts (hunk 3 above)
Comment (ledsun) [reply packages/npm-packages/ruby-wasm-wasi/src/browser.script.ts]: Fixed at https://github.com/ruby/ruby.wasm/pull/53/commits/19d373e05d153baba815420d584774aa3c2ed37e .
I used encodeURI with the intention of supporting filenames containing spaces.
I tried again and confirmed that it is not necessary.

File: packages/npm-packages/ruby-wasm-wasi/src/browser.script.ts (hunk 1 above)
Comment (kateinoigakukun) [reply packages/npm-packages/ruby-wasm-wasi/src/browser.script.ts:27]: Thanks so much!

**Review by kateinoigakukun — APPROVED** [review]
LGTM ❤️

**Pull request merged**